### PR TITLE
init fix for module with optional dependencies

### PR DESCRIPTION
**Pull request by amandadumi**
Addressing issue #65.

### CODE REVIEW

File: chemreps/__init__.py [hunk 1]
```diff
@@ -8,4 +8,9 @@
 from . import bat
 from . import just_bonds
 from . import dataset
-from . import fingerprints
+try:
+    import rdkit
+    from . import fingerprints
+except ImportError:
+   print('Warning: Optional dependency rdkit not installed, the fingerprints representation will not be available')
```
Comment (shivupa) [inline chemreps/__init__.py]: Can we raise an [`ImportWarning`](https://docs.python.org/3/library/exceptions.html#ImportWarning) with the same message which shouldn't crash the program?

Comment (amandadumi) [reply chemreps/__init__.py]: I originally had this was a warning;. However, I think the warning was caught in the init process and nothing was actually printing to screen in the notebook. I did want the user to be aware if fingerprints wasn’t working inside the notebook. Let me try once more though with the ImportWarning.

Comment (amandadumi) [reply chemreps/__init__.py]: Yeah never mind...I had made the error, i just forgot to actually raise the error...This is fixed now.

**Comment by shivupa** [comment]
Without `rdkit` the fingerprints test also breaks and the ImportWarning is interpreted like an error. Could we do something like [this](https://stackoverflow.com/a/14050282) to only import `rdkit` if it exists. Then the `fingerprints_test.py` needs to be changed to have the test skipped if `rdkit` can't be imported.

```
import numpy as np
import pytest as pt
from collections import OrderedDict
import pytest

def test_morganfp():
    pytest.importorskip("rdkit")
    from chemreps.fingerprints import morganfp
```

Once this goes in I think it will fix the documentation issues (or at least it does for me locally in combination with #64)

**Comment by amandadumi** [comment]
This is a nice fix, I wasn't aware of importerskip and was worried about testing this. Thank you.

**Comment by amandadumi** [comment]
However, we still have the issue of the user not knowing that fingerprints isn't imported, which I think would be nice. Would you agree? If so, just including and if statement for when the import returns `None` would be a simple solution that shouldn't break any tests.

**Comment by dlf57** [comment]
I think having that if statement would be good just in case! I am so sorry for not getting back on this

**Comment by amandadumi** [comment]
i had to update import lib call
from `rdkit_loader = importlib.find_loader('rdkit')`
to `rdkit_loader = importlib.util.find_spec('rdkit')` 
as the former was deprecated. 

Additionally, I added the ImportWarning to be ignored in coverage testing since I feel like our coverage is based on having all dependencies. Is this acceptable?
Either way, I did ensure correct behavior when RDKit is absent, which runs as below:

```
import chemreps
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/amandad/Documents/side_project/chemreps_stuff/chemreps/chemreps/__init__.py", line 16, in <module>
    raise ImportWarning('RDKit was not found. As a result, the Morgan fingerprint representation module has not been implemented.')
ImportWarning: RDKit was not found. As a result, the Morgan fingerprint representation module has not been implemented.
```

**Comment by dlf57** [comment]
This looks great. Is importlib a default python package or do we need to add it to the setup.py to pip install?

**Comment by amandadumi** [comment]
@dlf57 importlib is provided with python 3!

**Pull request merged**